offchain-modules/src/apps/rpc/handler.ts

```diff
@@ -0,0 +1,194 @@
+import { CkbTxGenerator } from '@force-bridge/ckb/tx-helper/generator';
+import { IndexerCollector } from '@force-bridge/ckb/tx-helper/collector';
+import { EthAsset } from '@force-bridge/ckb/model/asset';
+import { Amount } from '@lay2/pw-core';
+import { API, NervosNetwork, EthereumNetwork, AllNetworks } from './types';
+import { ForceBridgeCore } from '@force-bridge/core';
+import { Script } from '@lay2/pw-core';
+
+import { ethers } from 'ethers';
+import { abi } from '@force-bridge/xchain/eth/abi/ForceBridge.json';
+import { stringToUint8Array } from '@force-bridge/utils';
+import { JsonRpc } from 'eosjs';
+
+export class ForceBridgeAPIV1Handler implements API.ForceBridgeAPIV1 {
+  conn;
+  constructor(conn) {
+    this.conn = conn;
+  }
+  async generateBridgeInNervosTransaction(
+    payload: API.GenerateBridgeInTransactionPayload<EthereumNetwork>,
+  ): Promise<API.GenerateTransactionResponse<EthereumNetwork>> {
+    console.log('generateBridgeInNervosTransaction ', payload);
+
+    const sender = payload.sender;
+    const recipientLockscript = Script.fromRPC({
+      code_hash: payload.recipient.codeHash,
+      args: payload.recipient.args,
+      hash_type: payload.recipient.hashType,
+    });
+
+    const network = payload.asset.network;
+    let tx;
+    switch (network) {
+      case 'Ethereum':
+        const provider = new ethers.providers.JsonRpcProvider(ForceBridgeCore.config.eth.rpcUrl);
+        const bridgeContractAddr = ForceBridgeCore.config.eth.contractAddress;
+        const bridge = new ethers.Contract(bridgeContractAddr, abi, provider);
+        const sudtExtraData = '0x';
+        const ethAmount = ethers.utils.parseUnits(payload.asset.amount, 0);
+        const recipient = stringToUint8Array(recipientLockscript.toAddress().toCKBAddress());
+
+        switch (payload.asset.ident.address) {
+          case '0x0000000000000000000000000000000000000000':
```

Comment:
we can mark a `// TODO` here, to replace with  `EthereumModel.isNativeAsset(...)` when it is implemented